src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java

```diff
@@ -18,7 +18,8 @@
 public class TransactionDao {
 
     private static final String FIND_TRANSACTION_BY_EXTERNAL_ID = "SELECT * FROM transaction " +
-            "WHERE external_id = :externalId";
+            "WHERE external_id = :externalId " +
+            "AND (:gatewayAccountId is NULL OR gateway_account_id = :gatewayAccountId)";
```

Comment:
what does `:gatewayAccountId is NULL` mean?

Reply:
It means that it's not provided.

Reply:
👍